swesmith/profiles/base.py

```diff
@@ -0,0 +1,314 @@
+    # performed instead of running the entire test suite
+    # Affects valid.py
+    min_pregold: bool = False
+
+    def __init__(self):
+        """Initialize the repository profile with an instance-level lock."""
+        # NOTE on this design: the lock is to prevent concurrent clones of the same repository.
+        # In this repo, all subclasses of RepoProfile are meant to be Singletons (only one instance
+        # of the class will ever be created). If this changes for some reason in the future,
+        # this design may have to be updated.
+        self._lock = Lock()
+
+    @abstractmethod
+    def build_image(self):
+        """Build a Docker image (execution environment) for this repository profile."""
+        pass
+
+    @abstractmethod
+    def log_parser(self, log: str) -> dict[str, str]:
+        """Parse test output logs and extract relevant information."""
+        pass
+
+    @property
+    def image_name(self) -> str:
+        return f"{self.org_dh}/swesmith.{self.arch}.{self.owner}_1776_{self.repo}.{self.commit[:8]}".lower()
+
+    @property
+    def mirror_name(self):
+        return f"{self.org_gh}/{self.repo_name}"
+
+    @property
+    def repo_name(self):
+        return f"{self.owner}__{self.repo}.{self.commit[:8]}"
+
+    def _mirror_exists(self):
+        """Check if mirror repository exists under organization"""
+        return self.repo_name in [
+            x["name"]
+            for page in range(1, 3)  # TODO: Need to update over time
+            for x in api.repos.list_for_org(self.org_gh, per_page=100, page=page)
+        ]
+
+    def clone(self, dest: str | None = None) -> str | None:
+        """Clone repository locally"""
+        if not self._mirror_exists():
+            raise ValueError(
+                "Mirror clone repo must be created first (call .create_mirror)"
+            )
+        dest = self.repo_name if not dest else dest
+        if not os.path.exists(dest):
+            clone_cmd = (
+                f"git clone git@github.com:{self.mirror_name}.git"
+                if dest is None
+                else f"git clone git@github.com:{self.mirror_name}.git {dest}"
+            )
+            subprocess.run(
+                clone_cmd,
+                check=True,
+                shell=True,
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
+            return dest
+        return None
+
+    def create_mirror(self):
+        """Create a mirror of this repository at the specified commit."""
+        if self._mirror_exists():
+            return
+        if self.repo_name in os.listdir():
+            shutil.rmtree(self.repo_name)
+        api.repos.create_in_org(self.org_gh, self.repo_name)
+        for cmd in [
+            f"git clone git@github.com:{self.mirror_name}.git {self.repo_name}",
+            (
+                f"cd {self.repo_name}; "
+                f"git checkout {self.commit}; "
+                "rm -rf .git; "
```

Comment:
Just double checking that it's ok to not have the history in there/create a new commit ID. If we don't do this, we simply keep the same commit ID which might be easier.